simple/reprovide_test.go

```diff
@@ -95,6 +111,67 @@ func TestReprovide(t *testing.T) {
 	}
 }
 
+func TestTriggerTwice(t *testing.T) {
+	// Ensure we can only trigger once at a time.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	clA, _, _, _ := setupRouting(t)
+
+	keyCh := make(chan cid.Cid)
+	startCh := make(chan struct{})
+	keyFunc := func(ctx context.Context) (<-chan cid.Cid, error) {
+		<-startCh
+		return keyCh, nil
+	}
+
+	reprov := NewReprovider(ctx, time.Hour, clA, keyFunc)
+	go reprov.Run()
+	defer reprov.Close()
+
+	// Wait for the reprovider to start, otherwise, the reprovider will
+	// think a concurrent reprovide is running.
+	//
+	// We _could_ fix this race... but that would be complexity for nothing.
+	// 1. We start a reprovide 1 minute after startup anyways.
+	// 2. The window is really narrow.
+	time.Sleep(1 * time.Second)
+
+	errCh := make(chan error, 2)
+
+	// Trigger in the background
+	go func() {
+		errCh <- reprov.Trigger(ctx)
+	}()
+
+	// Wait for the trigger to really start.
+	startCh <- struct{}{}
+
+	// Try to trigger again, this should fail immediately.
+	if err := reprov.Trigger(ctx); err == nil {
```

Comment:
can we put a timer around this or other way to check that's it's failed quickly?